# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5MetaScannerCppSupplement.h
# Commit: 9a8abd87cc79
# Full Hash: 9a8abd87cc7935f29b94248c1a6f8203faa14403
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-08 21:43:06
# Regressor Bug: 1701828
# File Overlap Count: 10
# Description:
#   Bug 1701828 - meta charset rewrite. r=smaug
#   
#   Implements https://github.com/whatwg/html/issues/6962 . Improves performance
#   when <meta charset> occurs in head but after the first kilobyte and aligns
#   behavior better with WebKit and Blink.
# ==============================================================================

diff --git a/parser/html/nsHtml5MetaScannerCppSupplement.h b/parser/html/nsHtml5MetaScannerCppSupplement.h
deleted file mode 100644
--- a/parser/html/nsHtml5MetaScannerCppSupplement.h
+++ /dev/null
@@ -1,39 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "nsISupportsImpl.h"
-
-#include "mozilla/Encoding.h"
-
-const mozilla::Encoding* nsHtml5MetaScanner::sniff(nsHtml5ByteReadable* bytes) {
-  readable = bytes;
-  stateLoop(stateSave);
-  readable = nullptr;
-  return mEncoding;
-}
-
-bool nsHtml5MetaScanner::tryCharset(nsHtml5String charset) {
-  // This code needs to stay in sync with
-  // nsHtml5StreamParser::internalEncodingDeclaration. Unfortunately, the
-  // trickery with member fields here leads to some copy-paste reuse. :-(
-  nsAutoCString label;
-  nsString charset16;  // Not Auto, because using it to hold nsStringBuffer*
-  charset.ToString(charset16);
-  CopyUTF16toUTF8(charset16, label);
-  const mozilla::Encoding* encoding = Encoding::ForLabel(label);
-  if (!encoding) {
-    return false;
-  }
-  if (encoding == UTF_16BE_ENCODING || encoding == UTF_16LE_ENCODING) {
-    mEncoding = UTF_8_ENCODING;
-    return true;
-  }
-  if (encoding == X_USER_DEFINED_ENCODING) {
-    // WebKit/Blink hack for Indian and Armenian legacy sites
-    mEncoding = WINDOWS_1252_ENCODING;
-    return true;
-  }
-  mEncoding = encoding;
-  return true;
-}